python/lsst/ip/isr/overscan.py

```diff
@@ -154,10 +163,152 @@ def __init__(self, statControl=None, **kwargs):
     def run(self, exposure, amp, isTransposed=False):
         raise NotImplementedError("run method is not defined for OverscanCorrectionTaskBase")
 
-    def correctOverscan(self, exposure, amp, imageBBox, overscanBBox,
-                        isTransposed=True, leadingToSkip=0, trailingToSkip=0,
-                        overscanFraction=1.0, imageThreshold=np.inf,
-                        maskedRowColumnGrowSize=0):
+    @staticmethod
+    def _maskRowsOrColumns(
+        exposure,
+        overscanBBox,
+        overscanMaskedImage,
+        overscanMask,
+        maxDeviation,
+        maskedRowColumnGrowSize,
+        medianSmoothingKernel,
+        medianSmoothingOutlierThreshold,
+        doAbsoluteMaxDeviation,
+        isTransposed,
+    ):
+        """Mask overscan rows (~serial) or columns (~parallel).
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure containing the data.
+        overscanBBox: `lsst.geom.Box2I`
+            Bounding box for the overscan data.
+        overscanMaskedImage : `lsst.afw.image.MaskedImage`
+            Masked image containing the overscan data.
+        overscanMask : `np.ndarray`
+            Numpy array of the mask bits, anded with appropriate
+            mask planes.
+        maxDeviation : `float`
+            Maximum deviation from median (overscan units) to mask in overscan
+            correction. For parallel overscan this is a one-sided (positive
+            only) cut.
+        maskedRowColumnGrowSize : `int`
+            If a column (parallel overscan) or row (serial overscan) is
+            completely masked, then grow the mask by this radius. If the
+            value is <=0 then this will not be checked.
+        medianSmoothingKernel : `int`
+            Kernel (pixels) to smooth rows/columns. If <=0, median smoothing
+            is skipped. Otherwise must be odd.
+        medianSmoothingOutlierThreshold : `float`
+            Outlier threshold after median smoothing (overscan units). This
+            is applied only to positive outliers.
+        doAbsoluteMaxDeviation : `bool`
+            If true, deviation comparisons will use the absolute value;
+            otherwise it will cut positive outliers only.
+        isTransposed : `bool`
+            If true, then the data will be transposed before fitting
+            the overscan.
+
+        Returns
+        -------
+        badRowsColumns : `np.ndarray`
+            Array of bad rows (serial) or columns (parallel) that were
```

Comment:
These are the indices, right?

Reply:
Yes, and they've been renamed `badRowsOrColumns` as we discussed.